### PR TITLE
feat: api methods to get and store rewards options (WIN-148 + WIN-217)

### PR DESCRIPTION
**Pull request by blecorre**

### CODE REVIEW

File: swagger/swagger.yaml [hunk 1]
```diff
@@ -1912,6 +1912,161 @@ paths:
...
+        schema:
+          $ref: "#/components/schemas/UUID"
+    get:
+      responses:
+        200:
+          description: It's ok
+          content:
+            application/json:
+              schema:
+                type: array
```
Comment (Galeaf11) [inline swagger/swagger.yaml:1958]: please use $ref, for correct typing generation

File: swagger/swagger.yaml [hunk 2]
```diff
@@ -1912,6 +1912,161 @@ paths:
...
+        required: true
+        schema:
+          $ref: "#/components/schemas/UUID"
+    post:
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
```
Comment (Galeaf11) [inline swagger/swagger.yaml]: please use $ref, for correct typing generation

File: src/controllers/BookingController.ts [hunk 3]
```diff
@@ -40,6 +41,39 @@ export class BookingController {
...
+    req: AuthRequest,
+    res: Response,
+    next: NextFunction
+  ) {
+    try {
+      const { offerId } = req.params;
+
+      const data = await rewardService.getOptions(offerId);
+
+      res.json({ data, success: true });
```
Comment (Galeaf11) [inline src/controllers/BookingController.ts]: We abandoned data wrapper and status
just use `res.json(data)`

File: src/router/booking.ts [hunk 4]
```diff
@@ -9,4 +9,9 @@ export default (router: Router): void => {
     bookingController.myBookings
   );
   router.post('/booking/:offerId/guests', bookingController.setPassengers);
+  router.get(
```
Comment (Galeaf11) [inline src/router/booking.ts:12]: need to use `walletAuthMiddleware`, with wallet signature auth

File: src/router/booking.ts [hunk 5]
```diff
@@ -9,4 +9,9 @@ export default (router: Router): void => {
...
+    '/booking/:offerId/rewardOptions',
+    bookingController.getRewardOptions
+  );
+  router.post('/booking/:offerId/reward', bookingController.setRewardOption);
```
Comment (Galeaf11) [inline src/router/booking.ts]: need to use `walletAuthMiddleware`, with wallet signature auth

File: src/services/RewardService.ts [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    symbol: 'nct',
+    name: 'Toucan Protocol: Nature Carbon Tonne'
+  }
+};
+
+const rewardPercentage = 6;
+
+// Note: Free API is limited to 50 calls/minute.
+const apiURL = 'https://api.coingecko.com/api/v3/';
+const coinGeckoQuoteCurrencies = [
```
Comment (Galeaf11) [inline src/services/RewardService.ts]: now we have only 3 supported currencies ['EUR', 'USD', 'JPY'] 
just use `assetsCurrencies` from `@windingtree/win-commons/dist/types`

File: src/services/RewardService.ts [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    id: 'toucan-protocol-nature-carbon-tonne',
+    symbol: 'nct',
+    name: 'Toucan Protocol: Nature Carbon Tonne'
+  }
+};
+
+const rewardPercentage = 6;
+
+// Note: Free API is limited to 50 calls/minute.
+const apiURL = 'https://api.coingecko.com/api/v3/';
```
Comment (Galeaf11) [inline src/services/RewardService.ts]: move to .ENV and config

File: src/services/RewardService.ts [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+  'zar',
+  'xdr',
+  'xag',
+  'xau',
+  'sats'
+];
+
+export class RewardService {
+  public async getOptions(offerId: string): Promise<RewardOption[]> {
+    const offer = await offerRepository.getOne(offerId);
```
Comment (Galeaf11) [inline src/services/RewardService.ts]: don't use offers here, offers is temp data, lifetime of offer 30min, use deal, deal includes offer

File: src/services/RewardService.ts [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    const priceOffer = offer?.price?.public;
+    const currencyOffer = offer?.price?.currency;
+
+    if (!currencyOffer || !priceOffer) {
+      throw ApiError.NotFound('offer not found');
+    }
+
+    if (!coinGeckoQuoteCurrencies.includes(currencyOffer.toLowerCase())) {
```
Comment (Galeaf11) [inline src/services/RewardService.ts]: just use assetsCurrencies from @windingtree/win-commons/dist/types

File: src/types.ts [hunk 10]
```diff
@@ -124,3 +125,12 @@ export interface DealDTO {
 export type RouterInitializer = (router: Router) => void;
 
 export type HotelProviders = 'derbySoft' | 'amadeus';
+
+export type RewardTypes = undefined | 'CO2_OFFSET' | 'TOKEN';
```
Comment (Galeaf11) [inline src/types.ts]: remove undefined from `RewardTypes`

File: src/types.ts [hunk 11]
```diff
@@ -110,6 +110,7 @@ export interface DealDBValue {
   createdAt: Date;
   orderId?: string;
   message?: string;
+  rewardOption: RewardTypes;
```
Comment (Galeaf11) [inline src/types.ts]: rewardOption?: RewardTypes;
? after field = can be undifined

**Comment by Galeaf11** [comment]
also please add `rewardOption` to `DealDTO` type and to `bookingService.getDealsDTO` method
image

File: src/services/RewardService.ts (hunk 6 above)
Comment (blecorre) [reply src/services/RewardService.ts]: Yep, I know that, this check was to make sure that we are not calling them with a currency they don't support, but I decided to delete the check as their list can evolve.

File: src/config.ts [hunk 12]
```diff
@@ -104,3 +104,9 @@ export const providersUrls = {
 export const getUrlByKey = (key: HotelProviders): string => {
   return providersUrls[key];
 };
+
+// Rewards
+export const rewardPercentage = Number(process.env.REWARD_PERCENTAGE);
```
Comment (Galeaf11) [inline src/config.ts:113]: need to add this url fill check here:
image

**Review by Galeaf11 — APPROVED** [review]
LGTM

**Pull request merged**